### PR TITLE
Fix some missing categories in visual shader nodes

### LINKED ISSUE #89152: Visual shader warning "Clamp is missing a category" when opening a project and/or adding a node in visual shader editor.
### Tested versions

4.3 [01dc5c5b58e93cb893c9c427419eb7838e73ec7d]

### System information

Windows 11 - Vulkan - Nvidia RTX 4070 - intel i5 13600KF

### Issue description

When I add Clamp (perhaps some other nodes I haven't tested too) I get this warning, it also always appears when I open the project.

image

### Steps to reproduce

Add visual shader and add a Clamp method or just open the MRP.

### Minimal reproduction project (MRP)

[VisualShaderMRP.zip](https://github.com/godotengine/godot/files/14485865/VisualShaderMRP.zip)

### PR DESCRIPTION
**Pull request by AThousandShips**
Confirmed these with:
```diff
diff --git a/scene/resources/visual_shader.h b/scene/resources/visual_shader.h
index 09ea9a8890..3a7b3ea09c 100644
--- a/scene/resources/visual_shader.h
+++ b/scene/resources/visual_shader.h
@@ -366,7 +366,7 @@ public:

        virtual String get_warning(Shader::Mode p_mode, VisualShader::Type p_type) const;

-       virtual Category get_category() const;
+       virtual Category get_category() const = 0;

        VisualShaderNode();
 };
@@ -468,6 +468,8 @@ public:
        bool _is_highend() const;
        void _set_option_index(int p_op, int p_index);

+       virtual Category get_category() const override { return CATEGORY_SPECIAL; }
+
        int get_option_index(int p_op) const;
 };

```

* Fixes: https://github.com/godotengine/godot/issues/89152

### CODE REVIEW

File: scene/resources/visual_shader_sdf_nodes.h [hunk 1]
```diff
@@ -49,6 +49,8 @@ class VisualShaderNodeSDFToScreenUV : public VisualShaderNode {
 
 	virtual String generate_code(Shader::Mode p_mode, VisualShader::Type p_type, int p_id, const String *p_input_vars, const String *p_output_vars, bool p_for_preview = false) const override;
 
+	virtual Category get_category() const override { return CATEGORY_TEXTURE; }
```
Comment (AThousandShips) [inline scene/resources/visual_shader_sdf_nodes.h:52]: I'm unsure exactly what category to assign these to

File: scene/resources/visual_shader_particle_nodes.h [hunk 2]
```diff
@@ -75,8 +75,6 @@ class VisualShaderNodeParticleSphereEmitter : public VisualShaderNodeParticleEmi
 	virtual String generate_global_per_node(Shader::Mode p_mode, int p_id) const override;
 	virtual String generate_code(Shader::Mode p_mode, VisualShader::Type p_type, int p_id, const String *p_input_vars, const String *p_output_vars, bool p_for_preview = false) const override;
 
-	virtual Category get_category() const override { return CATEGORY_PARTICLE; }
```
Comment (AThousandShips) [inline scene/resources/visual_shader_particle_nodes.h:78]: These were redundant as the parent type already does this

File: scene/resources/visual_shader_nodes.h [hunk 3]
```diff
@@ -1990,8 +2010,6 @@ class VisualShaderNodeVectorDecompose : public VisualShaderNodeVectorBase {
 	virtual void set_op_type(OpType p_op_type) override;
 	virtual String generate_code(Shader::Mode p_mode, VisualShader::Type p_type, int p_id, const String *p_input_vars, const String *p_output_vars, bool p_for_preview = false) const override;
 
-	virtual Category get_category() const override { return CATEGORY_VECTOR; }
```
Comment (AThousandShips) [inline scene/resources/visual_shader_nodes.h:1993]: Was redundant

File: scene/resources/visual_shader_sdf_nodes.h [hunk 4]
```diff
@@ -49,6 +49,8 @@ class VisualShaderNodeSDFToScreenUV : public VisualShaderNode {
 
 	virtual String generate_code(Shader::Mode p_mode, VisualShader::Type p_type, int p_id, const String *p_input_vars, const String *p_output_vars, bool p_for_preview = false) const override;
 
+	virtual Category get_category() const override { return CATEGORY_TEXTURES; }
```
Comment (AThousandShips) [inline scene/resources/visual_shader_sdf_nodes.h:52]: I'm unsure exactly what category to assign these to

Comment (Geometror) [reply scene/resources/visual_shader_sdf_nodes.h:52]: We might want to add a new category for SDF nodes in the future. Or we just use CATEGORY_TEXTURES or CATEGORY_SPECIAL, at least that's what I would do for now.

**Review by Geometror — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank *you*!

**Pull request merged**